lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -907,8 +908,10 @@ def install_local(
     if dangerous:
         _cmd.append("--dangerous")
     try:
-        result = subprocess.check_output(_cmd, universal_newlines=True).splitlines()[0]
+        ansi_filter = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
+        result = subprocess.check_output(_cmd, universal_newlines=True).splitlines()[-1]
         snap_name, _ = result.split(" ", 1)
+        snap_name = ansi_filter.sub("", snap_name)
```

Comment:
A more general comment: it took you some debugging to pinpoint the failing line and the assumptions that didn't hold. How about wrapping some of the assumptions you're making here (the 'getitems', etc...) in try/except blocks, to raise some helpful exceptions and log some variables?

Reply:
Sounds like a good idea. I could probably add a top-level try-expect block that captures Snap API errors. We won't need to worry about the unpacking since the captured output piped to a list is being sliced correctly now, but we should make sure that the name is formatted correctly (i.e. no ANSI escape sequences).

Reply:
Added try-except block in https://github.com/canonical/operator-libs-linux/pull/56/commits/8b13e63cdd98108dfd70e978ad4db8a225985b94.